### PR TITLE
Add papertrail action

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +13,8 @@ GPG key.
 ### git-sign
 
 Use this action to create signed git artifacts:
-```markdown
+
+```yaml
```
Comment (alcaeus) [inline README.md:17]: 🤦‍♂️ Thanks!

File: papertrail/action.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    required: true
+  release_version:
+    description: "The release version"
+    required: false
+  filenames:
+    description: "Artifact filenames to include in the report, space-separated"
+    required: true
+  token:
+    description: "The GitHub token for the action"
+    required: true
```
Comment (alcaeus) [inline papertrail/action.yml:15]: Can we use `default: ${{ github.token }}` here or does that not work?

File: papertrail/action.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: "Papertrail Report"
+description: "Generate report for authorized publication on distribution channels"
+inputs:
+  product_name:
+    description: "Name of product"
+    required: true
+  release_version:
+    description: "The release version"
```
Comment (alcaeus) [inline papertrail/action.yml:8]:
```suggestion
    description: "The release version. If not provided, the github.ref_name variable will be used"
```

**Review by alcaeus — COMMENTED** [review]
I haven't yet tested it yet, but looks good after a first look.

I'm a little concerned how this will fit the workflow. Take PHP for example: triggering the release is done by a user, but the build artifacts are only built later after the release tag has been pushed by the automation. So, in order for us to include hashes for generated artifacts, we'd have to wait until after packaging, but we no longer have information about who triggered the release build at that point.

File: papertrail/action.yml (hunk 2 above)
Comment (blink1073) [reply papertrail/action.yml:15]: Not from within an action, "Note: This context property is set by the Actions runner, and is only available within the execution steps of a job."

**Comment by blink1073** [comment]
> I'm a little concerned how this will fit the workflow. Take PHP for example: triggering the release is done by a user, but the build artifacts are only built later after the release tag has been pushed by the automation. So, in order for us to include hashes for generated artifacts, we'd have to wait until after packaging, but we no longer have information about who triggered the release build at that point.

Yes, we have to run everything in a single workflow.  See https://github.com/mongodb/winkerberos/pull/76 for how that works with the `workflow_call`.

File: papertrail/action.yml (hunk 2 above)
Comment (alcaeus) [reply papertrail/action.yml:15]: I see. No problem then 👍

**Review by alcaeus — APPROVED** [review]

**Pull request merged**